torch/onnx/symbolic_opset11.py

```diff
@@ -180,6 +203,62 @@ def round(g, self):
     return g.op("Round", self)
 
 
+# Generate paddings in ONNX order based on pad in pytorch.
+# Arguments:
+#     dim: the dimension of the tensor.
+#     pad: the paddings in pytorch.
+#          The order is dim_n_begin, dim_n_end, dim_n-1_begin, dim_n-1_end, ..., dim_m_begin, dim_m_end,
+#          where m is in range [0, n].
+def _prepare_onnx_paddings(g, dim, pad):
+    # The desired order of paddings is
+    # dim_0_begin, dim_1_begin, ... , dim_0_end, ..., dim_n_end.
+    # n is the dimension of input.
+    # Assume zero-dimensions in the beginning, pad the "pad" sequence with zeros in the beginning
+    pad_len = torch.onnx.symbolic_opset9.size(g, pad, g.op("Constant", value_t=torch.tensor([0])))
+    # Set extension = [0] * (dim * 2 - len(pad))
+    extension = g.op("Sub", g.op("Mul", g.op("Constant", value_t=torch.tensor(dim, dtype=torch.int64)),
+                     g.op("Constant", value_t=torch.tensor(2, dtype=torch.int64))), pad_len)
+    # Concat pad with extension: paddings = [dim_n_begin, dim_n_end, dim_n-1_begin, dim_n-1_end, 0, 0, ... ]
+    paddings = g.op("Concat", pad, g.op("ConstantOfShape", extension, value_t=torch.tensor([0])), axis_i=0)
```

Comment:
I get this error from this line:

> Type parameter (T) bound to different types (tensor(int32) and tensor(int64) in node (__Concat_213).

Reply:
ONNX only supports int64 for now. I added the cast, but you cannot have int pads in ONNX for now.